core/trino-spi/src/main/java/io/trino/spi/ptf/ConnectorTableFunctionHandle.java

```diff
@@ -14,11 +14,13 @@
 package io.trino.spi.ptf;
 
 import io.trino.spi.Experimental;
+import io.trino.spi.connector.CatalogHandle;
 
 /**
  * An area to store all information necessary to execute the table function, gathered at analysis time
  */
 @Experimental(eta = "2022-10-31")
 public interface ConnectorTableFunctionHandle
 {
+    CatalogHandle getCatalogHandle();
```

Comment:
`*Handle` interfaces are generally opaque, marker interfaces.
This add more freedom when implementing them.
For example, a stateless implementation can use a singleton.

In this particular case, why do we expect ConnectorTableFunctionHandle to tell engine what is the CatalogHandle? How did the engine obtain ConnectorTableFunctionHandle? Did it know CatalogHandle already when doing so?